fe1-web/model/objects/Chirp.ts

```diff
@@ -0,0 +1,87 @@
+import { Timestamp } from './Timestamp';
+import { Hash } from './Hash';
+
+/**
+ * Object to represent a Chirp.
+ */
+
+export interface ChirpState {
+  id: string;
+  sender: string;
+  text: string;
+  time: number;
+  likes: number;
+  parentId: string;
+}
+
+export class Chirp {
+  public readonly id: Hash;
+
+  // The sender's public key
+  public readonly sender: string;
+
+  // The text of the chirp
+  public readonly text: string;
+
+  // The time where the chirp was posted
+  public readonly time: Timestamp;
+
+  // The number of likes
+  public readonly likes: number;
+
+  // The id of the parent chirp (if it is a reply)
+  public readonly parentId?: Hash;
+
+  constructor(obj: Partial<Chirp>) {
+    if (obj === undefined || obj === null) {
+      throw new Error('Error encountered while creating a Chirp object: '
+        + 'undefined/null parameters');
+    }
+
+    if (obj.id === undefined) {
+      throw new Error("Undefined 'id' when creating 'Chirp'");
+    }
+    if (obj.sender === undefined) {
+      throw new Error("Undefined 'sender' when creating 'Chirp'");
+    }
+    if (obj.text === undefined) {
+      throw new Error("Undefined 'text' when creating 'Chirp'");
+    }
+    if (obj.time === undefined) {
+      throw new Error("Undefined 'id' when creating 'Chirp'");
+    }
+    if (obj.likes === undefined) {
+      throw new Error("Undefined 'likes' when creating 'Chirp'");
+    }
```

Comment:
Do we have this information when creating the object the first time around? 
If not, 0 likes might be a reasonable default too.

Reply:
That's true !